// This file is part of DynExp.

syntax = "proto3";

import public "Common.proto";

package DynExpProto.NetworkAnalogIn;

message ValueLimitsMessage
{
	double LowerLimit = 1;
	double UpperLimit = 2;
}

message ValueResolutionMessage
{
	double Resolution = 1;
}

message ValueUnitMessage
{
	DynExpProto.Common.UnitType Unit = 1;
}

message SampleMessage
{
	double Value = 1;
}

service NetworkAnalogIn
{
	rpc GetHardwareLimits (DynExpProto.Common.VoidMessage) returns (ValueLimitsMessage) {}
	rpc GetHardwareResolution (DynExpProto.Common.VoidMessage) returns (ValueResolutionMessage) {}
	rpc GetValueUnit (DynExpProto.Common.VoidMessage) returns (ValueUnitMessage) {}
	rpc GetAsync (DynExpProto.Common.VoidMessage) returns (SampleMessage) {}
	rpc GetSync (DynExpProto.Common.VoidMessage) returns (SampleMessage) {}
}